src/test/groovy/org/prebid/server/functional/tests/module/GeneralModuleSpec.groovy

```diff
@@ -0,0 +1,239 @@
+        given: "Default bid request with verbose trace"
+        def bidRequest = defaultBidRequest.tap {
+            ext.prebid.trace = TraceLevel.VERBOSE
+        }
+
+        and: "Save account with enabled response correction module"
+        def pbsModulesConfig = new PbsModulesConfig(pbRichmediaFilter: pbRichmediaFilterConfig, pbResponseCorrection: pbResponseCorrectionConfig)
+        def accountConfig = new AccountConfig(hooks: new AccountHooksConfiguration(modules: pbsModulesConfig))
+        def account = new Account(uuid: bidRequest.getAccountId(), config: accountConfig)
+        accountDao.save(account)
+
+        and: "Flush metrics"
+        flushMetrics(pbsServiceWithMultipleModuleWithRequireInvoke)
+
+        when: "PBS processes auction request"
+        def response = pbsServiceWithMultipleModuleWithRequireInvoke.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should include trace information about called modules"
+        verifyAll(response?.ext?.prebid?.modules?.trace?.stages?.outcomes?.groups?.invocationResults?.flatten() as List<InvocationResult>) {
+            it.status == [SUCCESS, SUCCESS]
+            it.action == [NO_ACTION, NO_ACTION]
+            it.hookId.moduleCode.sort() == [PB_RICHMEDIA_FILTER, PB_RESPONSE_CORRECTION].code.sort()
+        }
+
+        and: "no-invocation metrics shouldn't be updated"
+        def metrics = pbsServiceWithMultipleModuleWithRequireInvoke.sendCollectedMetricsRequest()
+        assert !metrics[NO_INVOCATION_METRIC.formatted(PB_RICHMEDIA_FILTER.code, ALL_PROCESSED_BID_RESPONSES.metricValue, PB_RICHMEDIA_FILTER_ALL_PROCESSED_RESPONSES.code)]
+        assert !metrics[NO_INVOCATION_METRIC.formatted(PB_RESPONSE_CORRECTION.code, ALL_PROCESSED_BID_RESPONSES.metricValue, RESPONSE_CORRECTION_ALL_PROCESSED_RESPONSES.code)]
+
+        and: "hook call metrics should be updated"
+        assert metrics[CALL_METRIC.formatted(PB_RICHMEDIA_FILTER.code, ALL_PROCESSED_BID_RESPONSES.metricValue, PB_RICHMEDIA_FILTER_ALL_PROCESSED_RESPONSES.code)] == 1
+        assert metrics[CALL_METRIC.formatted(PB_RESPONSE_CORRECTION.code, ALL_PROCESSED_BID_RESPONSES.metricValue, RESPONSE_CORRECTION_ALL_PROCESSED_RESPONSES.code)] == 1
+
+        where:
+        pbRichmediaFilterConfig                | pbResponseCorrectionConfig
+        new RichmediaFilter()                  | new PbResponseCorrection()
+        new RichmediaFilter()                  | new PbResponseCorrection(enabled: false)
+        new RichmediaFilter()                  | new PbResponseCorrection(enabled: true)
+        new RichmediaFilter(filterMraid: true) | new PbResponseCorrection()
+        new RichmediaFilter(filterMraid: true) | new PbResponseCorrection(enabled: true)
+    }
+
+    def "PBS should call specified module and traces response when account config includes that module and require-config-to-invoke is enabled"() {
```

Comment:
Do we need the same test as the previous one?

Reply:
This case verifies the feature's ability to filter modules as needed. Should they be combined?